src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Extensions/DataSetVersionAuthExtensions.cs

```diff
@@ -20,4 +26,13 @@ public static IQueryable<DataSetVersion> WherePublicStatus(this IQueryable<DataS
 
     private static Expression<Func<DataSetVersion, bool>> IsPublicStatus()
         => dataSetVersion => PublicStatuses.Contains(dataSetVersion.Status);
+
+    public static bool IsPrivateStatus(this DataSetVersion dataSetVersion)
+        => IsPrivateStatus().Compile()(dataSetVersion);
```

Comment:
Yeah, it's to allow code to be shareable between normal service code and within LINQ queries I believe (@jack-hive having created the original example for the `IsPublicStatus()` method).  While the Linq / EF bridge code might otherwise complain when given a DataSetVersion.IsPrivateStatus() method that it can't compile the expression down to SQL, this provides a way to do that, and additionally allows you to use this method as an extension method on the domain model so you can say `dataSetVersion.IsPrivateStatus()`.

This way we're much more assured that the logic for determining public / private statuses in queries also matches the same logic as we use in normal code.

Reply:
Nice, I'll bear this one in mind, might come in handy in the future!